In [1]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

In [2]:
population = pd.read_csv("fr_population.csv")
cereal = pd.read_csv("fr_cereales.csv")
sous_alimentation = pd.read_csv('fr_sousalimentation.csv')
vegetal = pd.read_csv('fr_vegetaux.csv')
animal = pd.read_csv('fr_animaux.csv')

# Question 1 : Calculez le nombre d'humains sur terre 

In [3]:
world_population = 1000 * population["Valeur"].sum()
world_population

8413993000

La population mondiale de 2013 est largement supérieure à celle de nos jours ce qui est une erreur. On constate que la population de la chine est comptée en double. Suppression de la valeur de la population globale de la chine pour garder celle des sous-régions. 

In [4]:
indexChine = population[ population['Symbole'] == 'A' ].index
population.drop(indexChine, inplace=True)
nb_people = 1000 * population["Valeur"].sum()
nb_people

6997326000

Le résultat obtenu semble plus cohérent avec la population mondiale en 2013

# Question 2:

In [5]:
# Création d'un dataframe végétaux en fonction des pays et des produits
vegetal_produit = vegetal.pivot_table(index=['Zone','Produit'], columns='Élément', values=['Valeur'])
vegetal_produit.fillna(0, inplace=True)

#Disponibilité intérieure du blé en France
ble = vegetal_produit.loc['France','Blé']
dispo_int_ble_fr = ble['Valeur']['Disponibilité intérieure']
dispo_int_ble_fr

20298.0

In [6]:
dispo_int_ble_fr_2 = ble['Valeur']['Production'] + ble['Valeur']['Importations - Quantité'] - ble['Valeur']['Exportations - Quantité'] + ble['Valeur']['Variation de stock']
dispo_int_ble_fr_2

20298.0

In [7]:
population_pivot = population.pivot_table(index='Zone', columns='Élément', values='Valeur')
population_pivot['Population totale'] = population_pivot['Population totale'] * 1000
population_fr = population_pivot.loc[('France'),'Population totale']

dispo_int_ble_fr_3 = ble['Valeur']['Semences'] + ble['Valeur']['Pertes'] + ble['Valeur']['Aliments pour animaux'] + (((ble['Valeur']['Disponibilité alimentaire en quantité (kg/personne/an)']*population_fr)/1000)/1000) + ble['Valeur']['Traitement'] +ble['Valeur']['Autres utilisations (non alimentaire)']
dispo_int_ble_fr_3

20298.07313

On constate qu'à un chiffre après la virgule dispo_int_ble_fr = dispo_int_ble_fr_1 = dispo_int_ble_fr_2

# Création d'un Dataframe général

In [8]:
# création d'une colonne type dans les dataframes 
vegetal['Type'] = 'vegetal'
animal['Type'] = 'animal'
cereal['is_cereal'] = True

vegetal = vegetal[vegetal['Zone'] != 'Chine' ].copy()
animal = animal[animal['Zone'] != 'Chine'].copy()
population = population[population['Zone'] != 'Chine'].copy()

#Union des deux Dataframes animal et vegetal
main_df = pd.concat([animal, vegetal])
# Création d'un tableau croisé dynamique
main_df = main_df.pivot_table(index=['Zone','Produit','Code Produit', 'Type','Code zone', 'Année'], columns='Élément', values=['Valeur'] )
main_df = main_df.reset_index()
anim_vegetal = main_df.copy()
# Ajout du dataframe Population
main_df = main_df.merge(population, on='Zone', how="left")
#Ajout du dataframe cereal
# remplacer les valeurs manquantes par 0
main_df.fillna(0, inplace=True)
main_df['Code produit'] = main_df['Code Produit','']


# Création d'une colonne 'Population_totale'
main_df['Population_totale'] = main_df['Valeur']*1000
main_df = main_df.drop(['Zone', 'Code zone','Code Domaine', 'Domaine' ,'Valeur' ,'Code Élément', ('Année', ''), 'Élément', 'Code Produit', 'Produit', 'Code année', 'Unité', 'Symbole', 'Description du Symbole'], axis=1)


main_df

(Zone, )            (Produit, )  (Code Produit, ) (Type, )  \
0       Afghanistan       Abats Comestible              2736   animal   
1       Afghanistan        Agrumes, Autres              2614  vegetal   
2       Afghanistan  Aliments pour enfants              2680  vegetal   
3       Afghanistan                 Ananas              2618  vegetal   
4       Afghanistan                Bananes              2615  vegetal   
...             ...                    ...               ...      ...   
15600  Îles Salomon       Viande de Suides              2733   animal   
15601  Îles Salomon    Viande de Volailles              2734   animal   
15602  Îles Salomon          Viande, Autre              2735   animal   
15603  Îles Salomon                    Vin              2655  vegetal   
15604  Îles Salomon         Épices, Autres              2645  vegetal   

       (Code zone, )  (Valeur, Aliments pour animaux)  \
0                  2                              0.0   
1                  2                              0.0   
2                  2                              0.0   
3                  2                              0.0   
4                  2                              0.0   
...              ...                              ...   
15600             25                              0.0   
15601             25                              0.0   
15602             25                              0.0   
15603             25                              0.0   
15604             25                              0.0   

       (Valeur, Autres utilisations (non alimentaire))  \
0                                                  0.0   
1                                                  0.0   
2                                                  0.0   
3                                                  0.0   
4                                                  0.0   
...                                                ...   
15600                                              0.0   
15601                                              0.0   
15602                                              0.0   
15603                                              0.0   
15604                                              0.0   

       (Valeur, Disponibilité alimentaire (Kcal/personne/jour))  \
0                                                    5.0          
1                                                    1.0          
2                                                    1.0          
3                                                    0.0          
4                                                    4.0          
...                                                  ...          
15600                                               45.0          
15601                                               11.0          
15602                                                0.0          
15603                                                0.0          
15604                                                4.0          

       (Valeur, Disponibilité alimentaire en quantité (kg/personne/an))  \
0                                                   1.72                  
1                                                   1.29                  
2                                                   0.06                  
3                                                   0.00                  
4                                                   2.70                  
...                                                  ...                  
15600                                               4.70                  
15601                                               3.34                  
15602                                               0.06                  
15603                                               0.07                  
15604                                               0.48                  

       (Valeur, Disponibilité de matière grass

# Question 3:  Calculez (pour chaque pays et chaque produit) la disponibilité alimentaire en kcal puis en kg de protéines

In [9]:
# Disponibilité alimentaire kcal par pays 
main_df['disp_alim_kcal_pays'] = main_df['Valeur', 'Disponibilité alimentaire (Kcal/personne/jour)'] * main_df['Population_totale'] * 365

# Disponibilité alimentaire kg par pays 
main_df['disp_alim_kg_an'] = (main_df['Valeur', 'Disponibilité de protéines en quantité (g/personne/jour)'] / 1000)*365
main_df['disp_alim_kg_pays'] = main_df['disp_alim_kg_an'] * main_df['Population_totale']
main_df

(Zone, )            (Produit, )  (Code Produit, ) (Type, )  \
0       Afghanistan       Abats Comestible              2736   animal   
1       Afghanistan        Agrumes, Autres              2614  vegetal   
2       Afghanistan  Aliments pour enfants              2680  vegetal   
3       Afghanistan                 Ananas              2618  vegetal   
4       Afghanistan                Bananes              2615  vegetal   
...             ...                    ...               ...      ...   
15600  Îles Salomon       Viande de Suides              2733   animal   
15601  Îles Salomon    Viande de Volailles              2734   animal   
15602  Îles Salomon          Viande, Autre              2735   animal   
15603  Îles Salomon                    Vin              2655  vegetal   
15604  Îles Salomon         Épices, Autres              2645  vegetal   

       (Code zone, )  (Valeur, Aliments pour animaux)  \
0                  2                              0.0   
1                  2                              0.0   
2                  2                              0.0   
3                  2                              0.0   
4                  2                              0.0   
...              ...                              ...   
15600             25                              0.0   
15601             25                              0.0   
15602             25                              0.0   
15603             25                              0.0   
15604             25                              0.0   

       (Valeur, Autres utilisations (non alimentaire))  \
0                                                  0.0   
1                                                  0.0   
2                                                  0.0   
3                                                  0.0   
4                                                  0.0   
...                                                ...   
15600                                              0.0   
15601                                              0.0   
15602                                              0.0   
15603                                              0.0   
15604                                              0.0   

       (Valeur, Disponibilité alimentaire (Kcal/personne/jour))  \
0                                                    5.0          
1                                                    1.0          
2                                                    1.0          
3                                                    0.0          
4                                                    4.0          
...                                                  ...          
15600                                               45.0          
15601                                               11.0          
15602                                                0.0          
15603                                                0.0          
15604                                                4.0          

       (Valeur, Disponibilité alimentaire en quantité (kg/personne/an))  \
0                                                   1.72                  
1                                                   1.29                  
2                                                   0.06                  
3                                                   0.00                  
4                                                   2.70                  
...                                                  ...                  
15600                                               4.70                  
15601                                               3.34                  
15602                                               0.06                  
15603                                               0.07                  
15604                                               0.48                  

       (Valeur, Disponibilité de matière grass

## Question 4 :

In [10]:
# Calcul du ratio kcal/kg
main_df['ratio kcal/kg'] = (main_df['Valeur', 'Disponibilité alimentaire (Kcal/personne/jour)'] * main_df['Population_totale'] *365) /(main_df['Valeur', 'Nourriture']*1000000)

#poids total 
total_weight = main_df['Valeur', 'Disponibilité alimentaire en quantité (kg/personne/an)']* main_df['Population_totale']
# Calcul du pourcentage en protéine 
main_df['pourcentage_proteine'] = main_df['disp_alim_kg_pays']/total_weight
main_df.fillna(0, inplace=True)
main_df

(Zone, )            (Produit, )  (Code Produit, ) (Type, )  \
0       Afghanistan       Abats Comestible              2736   animal   
1       Afghanistan        Agrumes, Autres              2614  vegetal   
2       Afghanistan  Aliments pour enfants              2680  vegetal   
3       Afghanistan                 Ananas              2618  vegetal   
4       Afghanistan                Bananes              2615  vegetal   
...             ...                    ...               ...      ...   
15600  Îles Salomon       Viande de Suides              2733   animal   
15601  Îles Salomon    Viande de Volailles              2734   animal   
15602  Îles Salomon          Viande, Autre              2735   animal   
15603  Îles Salomon                    Vin              2655  vegetal   
15604  Îles Salomon         Épices, Autres              2645  vegetal   

       (Code zone, )  (Valeur, Aliments pour animaux)  \
0                  2                              0.0   
1                  2                              0.0   
2                  2                              0.0   
3                  2                              0.0   
4                  2                              0.0   
...              ...                              ...   
15600             25                              0.0   
15601             25                              0.0   
15602             25                              0.0   
15603             25                              0.0   
15604             25                              0.0   

       (Valeur, Autres utilisations (non alimentaire))  \
0                                                  0.0   
1                                                  0.0   
2                                                  0.0   
3                                                  0.0   
4                                                  0.0   
...                                                ...   
15600                                              0.0   
15601                                              0.0   
15602                                              0.0   
15603                                              0.0   
15604                                              0.0   

       (Valeur, Disponibilité alimentaire (Kcal/personne/jour))  \
0                                                    5.0          
1                                                    1.0          
2                                                    1.0          
3                                                    0.0          
4                                                    4.0          
...                                                  ...          
15600                                               45.0          
15601                                               11.0          
15602                                                0.0          
15603                                                0.0          
15604                                                4.0          

       (Valeur, Disponibilité alimentaire en quantité (kg/personne/an))  \
0                                                   1.72                  
1                                                   1.29                  
2                                                   0.06                  
3                                                   0.00                  
4                                                   2.70                  
...                                                  ...                  
15600                                               4.70                  
15601                                               3.34                  
15602                                               0.06                  
15603                                               0.07                  
15604                                               0.48                  

       (Valeur, Disponibilité de matière grass

## Question 5 : 

In [11]:
# Changer les valeurs "inf" en Nan puis en 0
main_df = main_df.replace(np.inf, 0)

df_5 = main_df.copy()
df_5 = df_5[df_5['pourcentage_proteine'] != 0]
df_5 = df_5.groupby(df_5['Produit','']).mean()
df_5 = df_5.sort_values(by=['pourcentage_proteine'], ascending=False)
df_5.head(20)

(Code Produit, )  (Code zone, )  \
(Produit, )                                                 
Graines Colza/Moutarde              2558.0      39.000000   
Soja                                2555.0     128.663866   
Arachides Decortiquees              2556.0     126.048485   
Pois                                2547.0     127.292857   
Légumineuses Autres                 2549.0     125.481013   
Haricots                            2546.0     125.459459   
Plantes Oleiferes, Autre            2570.0     144.343750   
Sésame                              2561.0     127.990099   
Viande, Autre                       2735.0     126.968553   
Abats Comestible                    2736.0     125.431034   
Aliments pour enfants               2680.0     125.548872   
Cephalopodes                        2766.0     121.617391   
Viande de Bovins                    2731.0     125.431034   
Viande d'Ovins/Caprins              2732.0     125.070175   
Viande de Volailles                 2734.0     125.431034   
Piments                             2641.0     132.824561   
Poissons Pelagiques                 2763.0     124.470588   
Graines de tournesol                2557.0     132.767442   
Poissons Marins, Autres             2764.0     122.981132   
Poivre                              2640.0     130.181818   

                          (Valeur, Aliments pour animaux)  \
(Produit, )                                                 
Graines Colza/Moutarde                         246.333333   
Soja                                           134.949580   
Arachides Decortiquees                           0.006061   
Pois                                            23.550000   
Légumineuses Autres                             41.303797   
Haricots                                        15.925676   
Plantes Oleiferes, Autre                        10.125000   
Sésame                                           0.000000   
Viande, Autre                                    0.352201   
Abats Comestible                                 5.798851   
Aliments pour enfants                            0.000000   
Cephalopodes                                     0.339130   
Viande de Bovins                                 0.051724   
Viande d'Ovins/Caprins                           0.105263   
Viande de Volailles                              0.000000   
Piments                                          0.000000   
Poissons Pelagiques                            110.588235   
Graines de tournesol                            13.139535   
Poissons Marins, Autres                          8.622642   
Poivre                                           0.000000   

                          (Valeur, Autres utilisations (non alimentaire))  \
(Produit, )                                                                 
Graines Colza/Moutarde                                           0.000000   
Soja                                                             5.319328   
Arachides Decortiquees                                           7.666667   
Pois                                                             0.200000   
Légumineuses Autres                                              5.164557   
Haricots                                                         0.398649   
Plantes Oleiferes, Autre                                         3.437500   
Sésame                                                           0.188119   
Viande, Autre                                                    0.094340   
Abats Comestible                                                 4.873563   
Aliments pour enfants                                            0.000000   
Cephalopodes                                                     0.217391   
Viande de Bovins                                                 0.505747   
Viande d'Ovins/Caprins                                           0.368421   
Viande de Volailles                                              7.229885   
Piments                        

Parmis les 20 aliments les plus riches en protéine on retrouve : 'Graines Colza/Moutarde', 'viande de bovin', 'Arachides Decortiquees', 'Pois', 'Légumineuses Autres'

In [12]:
df_5 = df_5.sort_values(by=['ratio kcal/kg'], ascending=False)
df_5.head(20)

(Code Produit, )  (Code zone, )  \
(Produit, )                                                
Huile d'Arachide                   2572.0     255.000000   
Huile de Tournesol                 2573.0     117.157895   
Huile de Germe de Maïs             2582.0      39.000000   
Huile de Soja                      2571.0     129.400000   
Huile de Palme                     2577.0     122.379310   
Huile de Colza&Moutarde            2574.0     118.764706   
Huile Graines de Coton             2575.0     150.428571   
Huile de Palmistes                 2576.0     171.400000   
Huile de Coco                      2578.0     136.538462   
Huile de Son de Riz                2581.0     110.000000   
Huil Plantes Oleif Autr            2586.0     114.069444   
Beurre, Ghee                       2740.0     124.274336   
Graisses Animales Crue             2737.0     127.069930   
Arachides Decortiquees             2556.0     126.048485   
Sésame                             2561.0     127.990099   
Feve de Cacao                      2633.0     124.563291   
Riz (Eq Blanchi)                   2805.0     125.431034   
Sucre non centrifugé               2541.0      98.923077   
Sorgho                             2518.0     135.352941   
Millet                             2517.0     134.377358   

                         (Valeur, Aliments pour animaux)  \
(Produit, )                                                
Huile d'Arachide                                0.000000   
Huile de Tournesol                              0.000000   
Huile de Germe de Maïs                          0.000000   
Huile de Soja                                   0.000000   
Huile de Palme                                  0.000000   
Huile de Colza&Moutarde                        41.176471   
Huile Graines de Coton                          0.000000   
Huile de Palmistes                              0.000000   
Huile de Coco                                   0.000000   
Huile de Son de Riz                             0.000000   
Huil Plantes Oleif Autr                         0.000000   
Beurre, Ghee                                    0.008850   
Graisses Animales Crue                          9.454545   
Arachides Decortiquees                          0.006061   
Sésame                                          0.000000   
Feve de Cacao                                   0.012658   
Riz (Eq Blanchi)                              193.068966   
Sucre non centrifugé                           30.153846   
Sorgho                                        134.588235   
Millet                                         49.622642   

                         (Valeur, Autres utilisations (non alimentaire))  \
(Produit, )                                                                
Huile d'Arachide                                                2.000000   
Huile de Tournesol                                             50.368421   
Huile de Germe de Maïs                                         20.500000   
Huile de Soja                                                 186.920000   
Huile de Palme                                                232.827586   
Huile de Colza&Moutarde                                       180.176471   
Huile Graines de Coton                                         11.142857   
Huile de Palmistes                                              9.000000   
Huile de Coco                                                  12.153846   
Huile de Son de Riz                                            17.000000   
Huil Plantes Oleif Autr                                         9.166667   
Beurre, Ghee                                                    0.309735   
Graisses Animales Crue                                         49.174825   
Arachides Decortiquees                                          7.666667   
Sésame                                                          0.188119   
Feve de Cacao                                                   1.791139   
Riz (Eq Blanchi

Parmis les 20 aliments les plus caloriques on retrouve : 'Arachides Decortiquees', 'Sésame', 'Feve de Cacao', 'Riz (Eq Blanchi)','Millet'

## Question 6 :

In [13]:
df_6 = main_df.copy()
df_6 = df_6[df_6['Type',''] == 'vegetal'].copy()
df_6['Valeur','Disponibilité intérieure'] = df_6['Valeur','Disponibilité intérieure'] * 1000 *1000
df_6['Disponibilité_interieur_kcal'] = df_6['Valeur','Disponibilité intérieure'] * df_6['ratio kcal/kg']
disp_world_kcal = df_6['Disponibilité_interieur_kcal'].sum()
disp_world_kcal

1.2312091265143276e+16

## Question 7 :

In [14]:
# estimation du besoin calorique d'un humain sur 1 an 
human_kcal_year = 2250 * 365

# calcul du nombre d'humains pouvant être nourris par an en termes de kcal
world_human_kcal = disp_world_kcal /human_kcal_year  
world_human_kcal

14991891951.46822

In [17]:
# poids moyen d'un être humain dans le monde
moy_weight_human = 62

# besoin moyen d'un être humain en protéine sur 1 an
human_prot_year = (0.8 / 1000) * moy_weight_human * 365 

# Calcul de la disponibilité intérieure végétale en protéine 
df_6['Disponibilité_intérieur_prot'] = df_6['Valeur','Disponibilité intérieure'] * df_6['pourcentage_proteine']

disp_world_prot = df_6['Disponibilité_intérieur_prot'].sum()

human_world_prot = disp_world_prot / human_prot_year
human_world_prot


16128801690.255905

In [18]:
# pourcentage par rapport à la population mondiale 
percentage_cal = (world_human_kcal *100) / nb_people
percentage_cal

214.2517291815219

environs 214 % de la population mondiale peut être nourrie en termes de kcal

In [19]:
percentage_prot = (human_world_prot * 100 ) / nb_people
percentage_prot

230.49950352828932

environs 230 % de la population peut être nourrie en termes de protéines 

## Question 8 :

Combien d'humains pourraient être nourris si toute la disponibilité alimentaire en produits végétaux, la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture ? Donnez les résultats en termes de calories, puis de protéines, et exprimez ensuite ces 2 résultats en pourcentage de la population mondiale

In [20]:
df_8 = df_6.copy()

# additionner les différentes valeurs 'disponibilité alimentaire', 'nourriture destinée aux animaux' et les 'Pertes' pour chaque pays
# créer une colonne pour la disponibilité alimentaire totale  en kg  par pays .

df_8['disp_alim_kg_total'] = df_8['Valeur','Disponibilité alimentaire en quantité (kg/personne/an)'] * df_8['Population_totale']

df_8['poids_alim_anim_pertes_total'] = df_8['disp_alim_kg_total'] + (df_8['Valeur','Aliments pour animaux'] * 1000000) + (df_8['Valeur','Pertes'] * 1000000)
df_8['poids_alim_anim_pertes_total_kcal'] = df_8['poids_alim_anim_pertes_total'] * df_8['ratio kcal/kg']
poids_alim_anim_pertes_total_kcal = df_8['poids_alim_anim_pertes_total_kcal'].sum()
disp_anim_loss_kcal_world = poids_alim_anim_pertes_total_kcal / human_kcal_year
disp_anim_loss_kcal_percentage = (disp_anim_loss_kcal_world * 100/ nb_people)
disp_anim_loss_kcal_percentage


159.66047343018025

Le pourcentage représente 159% de la population mondiale qui pourrait être nourrie en termes de calories

In [22]:
# résultat en termes de proteine 
df_8['poids_alim_anim_pertes_total_prot'] = df_8['poids_alim_anim_pertes_total'] * df_8['pourcentage_proteine']
poids_alim_anim_pertes_total_prot = df_8['poids_alim_anim_pertes_total_prot'].sum()
disp_anim_loss_prot_world = poids_alim_anim_pertes_total_prot / human_prot_year
percentage_prot_world = (disp_anim_loss_prot_world  * 100 / nb_people)
percentage_prot_world

161.34080674604547

Le pourcentage représente 161% de la population mondiale qui pourrait être nourrie en termes de protéines

## Question 9 :


Combien d'humains pourraient être nourris avec la disponibilité alimentaire mondiale ? Donnez les résultats en termes de calories, puis de protéines, et exprimez ensuite ces 2 résultats en pourcentage de la population mondiale.

In [23]:
# calculez la disponibilité alimentaire mondiale kcal 

df_9 = main_df.copy()

disp_alim_kcal_totale = df_9['disp_alim_kcal_pays'].sum()
disp_alim_kcal_world = disp_alim_kcal_totale /  human_kcal_year

disp_alim_kcal_world_percentage  = (disp_alim_kcal_world * 100)/ nb_people

disp_alim_kcal_world_percentage

128.14433570131726

cela représente 128 % de la population mondiale qui pourraient être nourri en terme de calories

In [24]:
# disponibilité alimentaire  mondiale prot
disp_alim_prot_totale = df_9['disp_alim_kg_pays'].sum()
disp_alim_prot_world = disp_alim_prot_totale /  human_prot_year
disp_alim_prot_world_percentage = (disp_alim_prot_world * 100 )/ nb_people
disp_alim_prot_world_percentage 

163.60897618362563

cela  représente 163% de la population qui pourrait nourrie en termes de protéines

## Question 10 :


A partir des données téléchargées qui concernent la sous-nutrition, répondez à cette question : Quelle proportion de la population mondiale est considérée comme étant en sous-nutrition ?

In [25]:
df_10 = sous_alimentation
df_10[df_10['Valeur'] == '<0.1'] = 0
df_10['Valeur'] = pd.to_numeric(df_10['Valeur'])
df_10.fillna(0, inplace = True)


df_10['population'] = df_10['Valeur'] * 1000000
df_10 = df_10[df_10['Année'] == '2012-2014']
df_10 = df_10[df_10['Zone'] != 'Chine']
population_mondial_sous_alim  = df_10['population'].sum()
population_mondial_sous_alim_percentage = (population_mondial_sous_alim * 100)/nb_people
population_mondial_sous_alim_percentage

10.628345742359295

Les personnes étant en sous alimentation représente environs 10% de la population mondiale

# Question 11 :

En ne prenant en compte que les céréales destinées à l'alimentation (humaine et animale), quelle proportion (en termes de poids) est destinée à l'alimentation animale ?

In [26]:
code_cerales = cereal['Code Produit'].unique()
df_11 = main_df[main_df['Code Produit',''].isin(code_cerales)]
poids_alim_anim = df_11['Valeur', 'Aliments pour animaux'].sum() *1000000
poids_nourriture = df_11['Valeur', 'Nourriture'].sum() *1000000
poids_totale = poids_alim_anim + poids_nourriture
alim_anim_percentage = (poids_alim_anim * 100)/poids_totale
alim_anim_percentage

45.91402568664604

45% est destiné à l'alimentation animale

# Question 12 : 


Sélectionnez parmi les données des bilans alimentaires les informations relatives aux pays dans lesquels la FAO recense des personnes en sous-nutrition

In [27]:
# pays avec des personnes en sous_nutrition 
df_12 = df_10.copy()
df_12 = df_12[df_12['population'] > 0]

# Sélection des informations relatives à ces pays
df_12_bis = main_df[main_df['Zone',''].isin(df_12['Zone']) ]
df_12_bis

(Zone, )            (Produit, )  (Code Produit, ) (Type, )  \
0      Afghanistan       Abats Comestible              2736   animal   
1      Afghanistan        Agrumes, Autres              2614  vegetal   
2      Afghanistan  Aliments pour enfants              2680  vegetal   
3      Afghanistan                 Ananas              2618  vegetal   
4      Afghanistan                Bananes              2615  vegetal   
...            ...                    ...               ...      ...   
15522     Éthiopie       Viande de Suides              2733   animal   
15523     Éthiopie    Viande de Volailles              2734   animal   
15524     Éthiopie          Viande, Autre              2735   animal   
15525     Éthiopie                    Vin              2655  vegetal   
15526     Éthiopie         Épices, Autres              2645  vegetal   

       (Code zone, )  (Valeur, Aliments pour animaux)  \
0                  2                              0.0   
1                  2                              0.0   
2                  2                              0.0   
3                  2                              0.0   
4                  2                              0.0   
...              ...                              ...   
15522            238                              0.0   
15523            238                              0.0   
15524            238                              0.0   
15525            238                              0.0   
15526            238                              0.0   

       (Valeur, Autres utilisations (non alimentaire))  \
0                                                  0.0   
1                                                  0.0   
2                                                  0.0   
3                                                  0.0   
4                                                  0.0   
...                                                ...   
15522                                              0.0   
15523                                              0.0   
15524                                              0.0   
15525                                              0.0   
15526                                              0.0   

       (Valeur, Disponibilité alimentaire (Kcal/personne/jour))  \
0                                                    5.0          
1                                                    1.0          
2                                                    1.0          
3                                                    0.0          
4                                                    4.0          
...                                                  ...          
15522                                                0.0          
15523                                                2.0          
15524                                                4.0          
15525                                                0.0          
15526                                                3.0          

       (Valeur, Disponibilité alimentaire en quantité (kg/personne/an))  \
0                                                   1.72                  
1                                                   1.29                  
2                                                   0.06                  
3                                                   0.00                  
4                                                   2.70                  
...                                                  ...                  
15522                                               0.02                  
15523                                               0.66                  
15524                                               1.20                  
15525                                               0.02                  
15526                                               0.29                  

       (Valeur, Disponibilité de matière grasse en quanti

Repérez les 15 produits les plus exportés par ce groupe de pays.

In [28]:
df_12_bis['Exportation'] = df_12_bis['Valeur','Exportations - Quantité']
df_12_bis['Importation'] = df_12_bis['Valeur','Importations - Quantité']

df_12_bis['Code produit']
expo = df_12_bis.groupby(by=[df_12_bis['Code produit'], df_12_bis['Produit', '']]).sum().reset_index()
exportation = expo.sort_values(by='Exportation',ascending=False)

exportation_15 = exportation.iloc[:15,:]
exportation_15.reset_index()
exportation_15


Code produit          (Produit, )  (Code Produit, )  (Code zone, )  \
38          2577       Huile de Palme            242238          11909   
2           2514                 Maïs            241344          12151   
9           2532               Manioc            227880          11238   
95          2805     Riz (Eq Blanchi)            269280          12151   
16          2542        Sucre Eq Brut            244032          12151   
0           2511                  Blé            241056          12151   
47          2605      Légumes, Autres            250080          12151   
52          2615              Bananes            237965          11438   
58          2625       Fruits, Autres            252000          12151   
96          2848   Lait - Excl Beurre            273408          12151   
22          2555                 Soja            227395          11491   
85          2763  Poissons Pelagiques            262485          12149   
45          2601              Tomates            247095          11914   
54          2617               Pommes            251232          12151   
48          2611  Oranges, Mandarines            248045          12123   

    (Valeur, Aliments pour animaux)  \
38                              0.0   
2                          262845.0   
9                           75733.0   
95                          32927.0   
16                             23.0   
0                           48158.0   
47                          37584.0   
52                            828.0   
58                             43.0   
96                          40082.0   
22                          12109.0   
85                          13087.0   
45                              0.0   
54                            238.0   
48                             11.0   

    (Valeur, Autres utilisations (non alimentaire))  \
38                                          24743.0   
2                                           46095.0   
9                                           34090.0   
95                                          17838.0   
16                                           6094.0   
0                                           10113.0   
47                                             59.0   
52                                            116.0   
58                                             62.0   
96                                           1556.0   
22                                            637.0   
85                                             40.0   
45                                             48.0   
54                                             17.0   
48                                             23.0   

    (Valeur, Disponibilité alimentaire (Kcal/personne/jour))  \
38                                             6777.0          
2                                             24842.0          
9                                              8523.0          
95                                            38160.0          
16                                            20218.0          
0                                             44378.0          
47                                             3553.0          
52                                             2154.0          
58                                             2442.0          
96                                            11010.0          
22                                              651.0          
85                                              889.0          
45                                              752.0          
54                                              594.0          
48                                              660.0          

    (Valeur, Disponibilité alimentaire en quantité (kg/personne/an))  \
38                                             281.48                  
2                                             2924.63                  
9                                             3082.01                  


Parmis les données des bilans alimentaires au niveau mondial, sélectionnez les 200 plus grandes importations de ces produits (1 importation = une quantité d'un produit donné importée par un pays donné

In [29]:

importation = df_12_bis[df_12_bis['Produit',''].isin(exportation_15['Produit',''])]

importation = importation.sort_values(by='Importation',ascending=False)
importation_200 = importation.iloc[:200,:]
importation_200

(Zone, )         (Produit, )  (Code Produit, ) (Type, )  \
3353   Chine, continentale                Soja              2555  vegetal   
3323   Chine, continentale              Manioc              2532  vegetal   
15071               Égypte                 Blé              2511  vegetal   
6226                  Inde      Huile de Palme              2577  vegetal   
3320   Chine, continentale  Lait - Excl Beurre              2848   animal   
...                    ...                 ...               ...      ...   
692        Arabie saoudite             Tomates              2601  vegetal   
708              Argentine             Bananes              2615  vegetal   
14176              Tunisie       Sucre Eq Brut              2542  vegetal   
679        Arabie saoudite              Pommes              2617  vegetal   
13807            Thaïlande     Légumes, Autres              2605  vegetal   

       (Code zone, )  (Valeur, Aliments pour animaux)  \
3353              41                           9530.0   
3323              41                          22868.0   
15071             59                           4860.0   
6226             100                              0.0   
3320              41                           1608.0   
...              ...                              ...   
692              194                              0.0   
708                9                              0.0   
14176            222                              0.0   
679              194                              0.0   
13807            216                              0.0   

       (Valeur, Autres utilisations (non alimentaire))  \
3353                                               0.0   
3323                                            7940.0   
15071                                              0.0   
6226                                            6782.0   
3320                                               0.0   
...                                                ...   
692                                                0.0   
708                                                0.0   
14176                                              0.0   
679                                                0.0   
13807                                              0.0   

       (Valeur, Disponibilité alimentaire (Kcal/personne/jour))  \
3353                                                35.0          
3323                                                 6.0          
15071                                             1175.0          
6226                                                31.0          
3320                                                58.0          
...                                                  ...          
692                                                 16.0          
708                                                 22.0          
14176                                              330.0          
679                                                 10.0          
13807                                               36.0          

       (Valeur, Disponibilité alimentaire en quantité (kg/personne/an))  \
3353                                                3.66                  
3323                                                1.91                  
15071                                             146.83                  
6226                                                1.30                  
3320                                               32.66                  
...                                                  ...                  
692                                                29.42                  
708                                                12.84                  
14176                                              33.88                  
679                                                10.46                  
13807                                              47.86                  

     

Groupez ces importations par produit, afin d'avoir une table contenant 1 ligne pour chacun des 15 produits. Ensuite, calculez pour chaque produit les 2 quantités suivantes : le ratio entre la quantité destinée aux "Autres utilisations" (Other uses) et la disponibilité intérieure. le ratio entre la quantité destinée à la nourriture animale et la quantité destinée à la nourriture (animale + humaine)

In [30]:
importation_by_product = importation_200.groupby(importation_200['Produit','']).sum()
importation_by_product

(Code Produit, )  (Code zone, )  \
(Produit, )                                            
Bananes                         10460            377   
Blé                            125550           6728   
Fruits, Autres                  18375           1069   
Huile de Palme                  38655           1911   
Lait - Excl Beurre              59808           3073   
Légumes, Autres                 10420            673   
Manioc                          12660            703   
Maïs                            65364           3094   
Oranges, Mandarines              2611            194   
Poissons Pelagiques             16578            812   
Pommes                           5234            396   
Riz (Eq Blanchi)                67320           3447   
Soja                            25550           1553   
Sucre Eq Brut                   58466           3097   
Tomates                          5202            353   

                     (Valeur, Aliments pour animaux)  \
(Produit, )                                            
Bananes                                          0.0   
Blé                                          44858.0   
Fruits, Autres                                   0.0   
Huile de Palme                                   0.0   
Lait - Excl Beurre                            3600.0   
Légumes, Autres                                  1.0   
Manioc                                       25152.0   
Maïs                                        215133.0   
Oranges, Mandarines                              0.0   
Poissons Pelagiques                          10836.0   
Pommes                                           0.0   
Riz (Eq Blanchi)                             15907.0   
Soja                                         10668.0   
Sucre Eq Brut                                    0.0   
Tomates                                          0.0   

                     (Valeur, Autres utilisations (non alimentaire))  \
(Produit, )                                                            
Bananes                                                         21.0   
Blé                                                           9892.0   
Fruits, Autres                                                  41.0   
Huile de Palme                                               17886.0   
Lait - Excl Beurre                                            1326.0   
Légumes, Autres                                                 47.0   
Manioc                                                       21068.0   
Maïs                                                         40411.0   
Oranges, Mandarines                                              0.0   
Poissons Pelagiques                                             40.0   
Pommes                                                           0.0   
Riz (Eq Blanchi)                                             10403.0   
Soja                                                           524.0   
Sucre Eq Brut                                                 3509.0   
Tomates                                                          0.0   

                     (Valeur, Disponibilité alimentaire (Kcal/personne/jour))  \
(Produit, )                                                                     
Bananes                                                          113.0          
Blé                                                            29944.0          
Fruits, Autres                                                   195.0          
Huile de Palme                                                  1135.0          
Lait - Excl Beurre                                              2094.0          
Légumes, Autres                                                  150.0          
Manioc                                                           184.0          
Maïs                                                            5944.0          
Oranges, Mandarines                                                8.0          

 calculer le ratio entre la quantité destinée aux "Autres utilisations" (Other uses) et la disponibilité intérieure.

In [31]:
importation_by_product['ratio_other_disp'] = importation_by_product['Valeur','Autres utilisations (non alimentaire)'] / importation_by_product['Valeur','Disponibilité intérieure']
importation_by_product

(Code Produit, )  (Code zone, )  \
(Produit, )                                            
Bananes                         10460            377   
Blé                            125550           6728   
Fruits, Autres                  18375           1069   
Huile de Palme                  38655           1911   
Lait - Excl Beurre              59808           3073   
Légumes, Autres                 10420            673   
Manioc                          12660            703   
Maïs                            65364           3094   
Oranges, Mandarines              2611            194   
Poissons Pelagiques             16578            812   
Pommes                           5234            396   
Riz (Eq Blanchi)                67320           3447   
Soja                            25550           1553   
Sucre Eq Brut                   58466           3097   
Tomates                          5202            353   

                     (Valeur, Aliments pour animaux)  \
(Produit, )                                            
Bananes                                          0.0   
Blé                                          44858.0   
Fruits, Autres                                   0.0   
Huile de Palme                                   0.0   
Lait - Excl Beurre                            3600.0   
Légumes, Autres                                  1.0   
Manioc                                       25152.0   
Maïs                                        215133.0   
Oranges, Mandarines                              0.0   
Poissons Pelagiques                          10836.0   
Pommes                                           0.0   
Riz (Eq Blanchi)                             15907.0   
Soja                                         10668.0   
Sucre Eq Brut                                    0.0   
Tomates                                          0.0   

                     (Valeur, Autres utilisations (non alimentaire))  \
(Produit, )                                                            
Bananes                                                         21.0   
Blé                                                           9892.0   
Fruits, Autres                                                  41.0   
Huile de Palme                                               17886.0   
Lait - Excl Beurre                                            1326.0   
Légumes, Autres                                                 47.0   
Manioc                                                       21068.0   
Maïs                                                         40411.0   
Oranges, Mandarines                                              0.0   
Poissons Pelagiques                                             40.0   
Pommes                                                           0.0   
Riz (Eq Blanchi)                                             10403.0   
Soja                                                           524.0   
Sucre Eq Brut                                                 3509.0   
Tomates                                                          0.0   

                     (Valeur, Disponibilité alimentaire (Kcal/personne/jour))  \
(Produit, )                                                                     
Bananes                                                          113.0          
Blé                                                            29944.0          
Fruits, Autres                                                   195.0          
Huile de Palme                                                  1135.0          
Lait - Excl Beurre                                              2094.0          
Légumes, Autres                                                  150.0          
Manioc                                                           184.0          
Maïs                                                            5944.0          
Oranges, Mandarines                                                8.0          

le ratio entre la quantité destinée à la nourriture animale et la quantité destinée à la nourriture (animale + humaine)


In [32]:
importation_by_product['ratio_anim_nourriture'] = importation_by_product['Valeur','Aliments pour animaux'] / (importation_by_product['Valeur','Aliments pour animaux'] + importation_by_product['Valeur','Nourriture'] )
importation_by_product

(Code Produit, )  (Code zone, )  \
(Produit, )                                            
Bananes                         10460            377   
Blé                            125550           6728   
Fruits, Autres                  18375           1069   
Huile de Palme                  38655           1911   
Lait - Excl Beurre              59808           3073   
Légumes, Autres                 10420            673   
Manioc                          12660            703   
Maïs                            65364           3094   
Oranges, Mandarines              2611            194   
Poissons Pelagiques             16578            812   
Pommes                           5234            396   
Riz (Eq Blanchi)                67320           3447   
Soja                            25550           1553   
Sucre Eq Brut                   58466           3097   
Tomates                          5202            353   

                     (Valeur, Aliments pour animaux)  \
(Produit, )                                            
Bananes                                          0.0   
Blé                                          44858.0   
Fruits, Autres                                   0.0   
Huile de Palme                                   0.0   
Lait - Excl Beurre                            3600.0   
Légumes, Autres                                  1.0   
Manioc                                       25152.0   
Maïs                                        215133.0   
Oranges, Mandarines                              0.0   
Poissons Pelagiques                          10836.0   
Pommes                                           0.0   
Riz (Eq Blanchi)                             15907.0   
Soja                                         10668.0   
Sucre Eq Brut                                    0.0   
Tomates                                          0.0   

                     (Valeur, Autres utilisations (non alimentaire))  \
(Produit, )                                                            
Bananes                                                         21.0   
Blé                                                           9892.0   
Fruits, Autres                                                  41.0   
Huile de Palme                                               17886.0   
Lait - Excl Beurre                                            1326.0   
Légumes, Autres                                                 47.0   
Manioc                                                       21068.0   
Maïs                                                         40411.0   
Oranges, Mandarines                                              0.0   
Poissons Pelagiques                                             40.0   
Pommes                                                           0.0   
Riz (Eq Blanchi)                                             10403.0   
Soja                                                           524.0   
Sucre Eq Brut                                                 3509.0   
Tomates                                                          0.0   

                     (Valeur, Disponibilité alimentaire (Kcal/personne/jour))  \
(Produit, )                                                                     
Bananes                                                          113.0          
Blé                                                            29944.0          
Fruits, Autres                                                   195.0          
Huile de Palme                                                  1135.0          
Lait - Excl Beurre                                              2094.0          
Légumes, Autres                                                  150.0          
Manioc                                                           184.0          
Maïs                                                            5944.0          
Oranges, Mandarines                                                8.0          

Donnez les 3 produits qui ont la plus grande valeur pour chacun des 2 ratios (vous aurez donc 6 produits à citer)

In [33]:
top_ratio_anim = importation_by_product.sort_values(by='ratio_anim_nourriture', ascending=False)
top_ratio_anim.head(3)

(Code Produit, )  (Code zone, )  \
(Produit, )                                            
Maïs                            65364           3094   
Poissons Pelagiques             16578            812   
Soja                            25550           1553   

                     (Valeur, Aliments pour animaux)  \
(Produit, )                                            
Maïs                                        215133.0   
Poissons Pelagiques                          10836.0   
Soja                                         10668.0   

                     (Valeur, Autres utilisations (non alimentaire))  \
(Produit, )                                                            
Maïs                                                         40411.0   
Poissons Pelagiques                                             40.0   
Soja                                                           524.0   

                     (Valeur, Disponibilité alimentaire (Kcal/personne/jour))  \
(Produit, )                                                                     
Maïs                                                            5944.0          
Poissons Pelagiques                                              126.0          
Soja                                                             251.0          

                     (Valeur, Disponibilité alimentaire en quantité (kg/personne/an))  \
(Produit, )                                                                             
Maïs                                                            692.55                  
Poissons Pelagiques                                              50.76                  
Soja                                                             32.24                  

                     (Valeur, Disponibilité de matière grasse en quantité (g/personne/jour))  \
(Produit, )                                                                                    
Maïs                                                             63.15                         
Poissons Pelagiques                                               5.34                         
Soja                                                             11.44                         

                     (Valeur, Disponibilité de protéines en quantité (g/personne/jour))  \
(Produit, )                                                                               
Maïs                                                            151.33                    
Poissons Pelagiques                                              18.15                    
Soja                                                             25.24                    

                     (Valeur, Disponibilité intérieure)  \
(Produit, )                                               
Maïs                                           334220.0   
Poissons Pelagiques                             16419.0   
Soja                                            89821.0   

                     (Valeur, Exportations - Quantité)  ...  \
(Produit, )                                             ...   
Maïs                                            1333.0  ...   
Poissons Pelagiques                             3072.0  ...   
Soja                                             308.0  ...   

                     Population_totale  disp_alim_kcal_pays  disp_alim_kg_an  \
(Produit, )                                                                    
Maïs                        2563515000         1.465291e+14         55.23545   
Poissons Pelagiques         1925294000         5.065151e+12          6.62475   
Soja                        2091385000         2.237415e+13          9.21260   

                     disp_alim_kg_pays  ratio kcal/kg  pourcentage_proteine  \
(Produit, )                                                                   
Maïs                      3.625003e+09   78304.391586              1.979151   
Poissons Pelagiques       7.328422e+08    5628.836072      

les 3 aliments pour le ratio nourriture destiné aux animaux et la quantité destinée à la nourriture  sont : Maïs , Poissons Pelagique et Soja

In [34]:
top_ratio_other = importation_by_product.sort_values(by='ratio_other_disp', ascending=False)
top_ratio_other.head(3)

(Code Produit, )  (Code zone, )  \
(Produit, )                                       
Huile de Palme             38655           1911   
Manioc                     12660            703   
Maïs                       65364           3094   

                (Valeur, Aliments pour animaux)  \
(Produit, )                                       
Huile de Palme                              0.0   
Manioc                                  25152.0   
Maïs                                   215133.0   

                (Valeur, Autres utilisations (non alimentaire))  \
(Produit, )                                                       
Huile de Palme                                          17886.0   
Manioc                                                  21068.0   
Maïs                                                    40411.0   

                (Valeur, Disponibilité alimentaire (Kcal/personne/jour))  \
(Produit, )                                                                
Huile de Palme                                             1135.0          
Manioc                                                      184.0          
Maïs                                                       5944.0          

                (Valeur, Disponibilité alimentaire en quantité (kg/personne/an))  \
(Produit, )                                                                        
Huile de Palme                                              47.90                  
Manioc                                                      63.97                  
Maïs                                                       692.55                  

                (Valeur, Disponibilité de matière grasse en quantité (g/personne/jour))  \
(Produit, )                                                                               
Huile de Palme                                             128.33                         
Manioc                                                       0.38                         
Maïs                                                        63.15                         

                (Valeur, Disponibilité de protéines en quantité (g/personne/jour))  \
(Produit, )                                                                          
Huile de Palme                                               0.16                    
Manioc                                                       0.93                    
Maïs                                                       151.33                    

                (Valeur, Disponibilité intérieure)  \
(Produit, )                                          
Huile de Palme                             26450.0   
Manioc                                     66296.0   
Maïs                                      334220.0   

                (Valeur, Exportations - Quantité)  ...  Population_totale  \
(Produit, )                                        ...                      
Huile de Palme                            20231.0  ...         3741856000   
Manioc                                    26161.0  ...         1755491000   
Maïs                                       1333.0  ...         2563515000   

                disp_alim_kcal_pays  disp_alim_kg_an  disp_alim_kg_pays  \
(Produit, )                                                               
Huile de Palme         7.151876e+13          0.05840       7.063217e+06   
Manioc                 1.611638e+13          0.33945       9.298549e+07   
Maïs                   1.465291e+14         55.23545       3.625003e+09   

                ratio kcal/kg  pourcentage_proteine  Exportation  Importation  \
(Produit, )                                                                     
Huile de Palme  102838.111763              0.009931      20231.0      26634.0   
Manioc            4363.234913              0.026082      26161.0      33566.0   
Maïs             78304.391586              1.979151       1333.0      56510.0   

                ratio_other_disp  ratio_

les 3 aliments pour le ratio entre la quantité destinée aux "Autres utilisations" (Other uses) et la disponibilité intérieure sont : Huile de palme , Manioc et Maïs

# Question 13 :


Combien de tonnes de céréales pourraient être libérées si les USA diminuaient leur production de produits animaux de 10% ?

In [35]:
test = df_11[df_11['Zone',''] == "États-Unis d'Amérique"]
poid_total_anim_usa = test['Valeur', 'Aliments pour animaux'].sum() 
(poid_total_anim_usa * 0.1) * 1000

14009600.0

cela pourrait libérer 14009600 tonnes de céreales

# Question 14 :

En Thaïlande, quelle proportion de manioc est exportée ? Quelle est la proportion de personnes en sous-nutrition?

In [36]:
thailande_info = df_12_bis[df_12_bis['Zone',''] == 'Thaïlande']
thailande_info_manioc = df_12_bis[(df_12_bis['Zone',''] == 'Thaïlande') &  (df_12_bis['Produit',''] == 'Manioc')]
thailande_info_disp_manioc = thailande_info_manioc['Valeur','Production']
thailande_info_disp_manioc

13809    30228.0
Name: (Valeur, Production), dtype: float64

In [37]:
export_manioc = (thailande_info_manioc['Exportation'] * 100) / thailande_info_disp_manioc 
export_manioc

13809    83.41273
dtype: float64

environ 83% de sa production de manioc

In [38]:
sous_nutrition_thai = df_10[df_10['Zone'] == 'Thaïlande']
sous_nutrition_thai_pop = sous_nutrition_thai['population'] 
thai = sous_nutrition_thai.iloc[0]
thai.population

5600000.0

In [39]:
thai_pop = population[population['Zone'] == 'Thaïlande']
thai_pop_value = thai_pop['Valeur'] * 1000
thai_pop_value

160    67011000
Name: Valeur, dtype: int64

In [40]:
sous_nutrition_thai_percentage = (thai.population * 100) / thai_pop_value
sous_nutrition_thai_percentage

160    8.356837
Name: Valeur, dtype: float64

cela représente environ 8% de la population thaïlandaise